### ISSUE TITLE
[Feature][Discussion] RegExp supports

### ISSUE THREAD

**Issue by Tjatse** [Unitech/PM2#1528]
I find this feature is very helpful in a maintaining mode, especially that my app names are generated by
`
 "FIXED-NAME-" + process.env.SOME_ARGS
` 
i.e.:

| App name | id | mode |
| ------------ | -----|-------- |
| test-1 | 0 | cluster |
| test-1 | 1 | cluster | 
| test-2 | 2 | cluster |
| test-2 | 3 | cluster |
| test-3 | 4 | cluster |
| test-3 | 5 | cluster |

If I wanna restart them, it should be executed like following:
```
pm2 restart test-1
pm2 restart test-2
pm2 restart test-3
...
```

It's too complicated, not simple enough, maybe we need a command like:
```
pm2 reload /^test-\d+$/i
```
And this will be very useful in a condition like restarting processes by indices:
```
# restart by ids: 0-2
pm2 restart /[0-2]/
# restart by ids: 100-129
pm2 restart /^1[012][0-9]$/
```

What do you think?

**Comment by jshkurti** [comment]
Why not :)
You wanna do this ?

**Comment by Tjatse** [comment]
Yup, if you guys think it is doable, it will be an interesting feature.

**Comment by jshkurti** [comment]
It is doable for sure, the thing is, it requires a lot of refactoring.

**Comment by Tjatse** [comment]
Indeed, but it's worth to try, and in an elegant way o.~
